java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -972,26 +974,168 @@ public Description matchVariable(VariableTree tree, VisitorState state) {
       boolean containsFlagName =
           enumConstructorArgsContainsFlagName(nct.getArguments(), enumRecords);
       if (containsFlagName) {
-        return removeEnumValue(tree, state, state.getSourceForNode(tree));
+        return removeEnumValue(
+            tree, state, state.getSourceForNode(tree), isOnlyEnumConstant(sym, state));
       }
     }
 
+    // We also match the enum constant previous to the one being removed in some cases,
+    // to fix delimiters.
+    if (sym != null && sym.isEnum()) {
+      EnumEnding enumEnding = getEndingOfLastEnumConstantIfRemoved(sym, state);
+
+      if (enumEnding == EnumEnding.IGNORE) {
+        return Description.NO_MATCH;
+      }
+
+      // The next enum constant in the list will be removed by Piranha
+      // Let's replace the current enum constant's ending with the previous one
+      return buildDescription(tree)
+          .addFix(
+              SuggestedFix.replace(tree, state.getSourceForNode(tree) + enumEnding.getChar(), 0, 1))
+          .build();
+    }
+
     return Description.NO_MATCH;
   }
 
+  /** Represents the character at the end of an enum constant */
+  enum EnumEnding {
+    // Ignore the result of the trailing enum character, whether we know what it is or not
+    IGNORE,
+    COMMA(","),
+    SEMICOLON(";"),
+    NONE("");
+    @Nullable final String endingChar;
+
+    @Nullable
+    String getChar() {
+      return endingChar;
+    }
+
+    EnumEnding() {
+      this.endingChar = null;
+    }
+
+    EnumEnding(@Nullable final String endingChar) {
```

Comment:
The argument doesn't need to be `@Nullable`, since we have the other constructor for that case, just leave it as `final String endingChar` (which is by convention non-null, and will be enforced as such if we turn on NullAway for this code). You can assign non-null to `@Nullable`, just not viceversa.

Reply:
Whatever settings I have in IntelliJ isn't too happy if I remove it (I get a `Constructor parameter for @Nullable field might be annotated @Nullable itself ` warning), so that's why it's even there. It can't tell that we use the other constructor for the null case.

It might be more explicit to remove the other constructor and pass `null` anyway? So I will remove the default constructor and pass null as the `endingChar`, since it's less indirection when reading the code.

Reply:
Ok, that works for me, but that warning sounds to me like more of a code suggestion than something that must be addressed, it's perfectly reasonable to have a constructor with a non-null argument assign it to a nullable field and have a different constructor that leaves the field null...